src/shared/hooks/useDisablePictureInPicture.tsx

```diff
@@ -0,0 +1,20 @@
+import { type RefObject, useEffect } from 'react';
+
+export const useDisablePictureInPicture = (ref: RefObject<HTMLElement>) => {
+	useEffect(() => {
+		if (ref.current) {
+			observer.observe(ref.current, {
+				childList: true,
+				subtree: true,
+			});
+		}
+
+		return () => observer.disconnect();
+	});
+
+	const observer = new MutationObserver(() => {
+		Array.from(document.getElementsByTagName('video')).forEach((element) => {
+			element.disablePictureInPicture = true;
+		});
+	});
```

Comment:
Seems more logical to define the observer const first, then use it in the useEffect
```suggestion
	const observer = new MutationObserver(() => {
		Array.from(document.getElementsByTagName('video')).forEach((element) => {
			element.disablePictureInPicture = true;
		});
	});

	useEffect(() => {
		if (ref.current) {
			observer.observe(ref.current, {
				childList: true,
				subtree: true,
			});
		}

		return () => observer.disconnect();
	});
```